poly/src/domain/general.rs

```diff
@@ -33,6 +37,123 @@ macro_rules! map {
     }
 }
 
+impl<F: FftField> CanonicalSerialize for GeneralEvaluationDomain<F> {
+    fn serialize<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        let type_id = match self {
+            GeneralEvaluationDomain::Radix2(_) => 0u8,
+            GeneralEvaluationDomain::MixedRadix(_) => 1u8,
+        };
+        type_id.serialize(&mut writer)?;
+
+        match self {
+            GeneralEvaluationDomain::Radix2(d) => d.serialize(&mut writer),
```

Comment:
Could we instead add a serialization method on the underlying domains, instead of serializing the degree directly?

Reply:
oh I see the underlying methods the `d` here is domain, not `degree`. Can the variable be changed throughout the file to be `domain` instead of `d`

Reply:
Yes. I can do the change. I also have implemented the serialization for domains.

The serialization of domains is needed to serialize the evaluations.